provider/kubernetes/kubernetes.go

```diff
@@ -219,8 +219,13 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 					}
 				}
 
-				if rule := getRuleForPath(pa, i); rule != "" {
-					templateObjects.Frontends[baseName].Routes[pa.Path] = types.Route{
+				rule, err := getRuleForPath(pa, i)
+				if err != nil {
+					log.Errorf("fail to get rule from ingress path: %s", err.Error())
+					return nil, err
```

Comment:
Returning an error would cause all Ingress processing to end prematurely, leaving the old configuration active. Thus, a single misconfigured Ingress could bring everything to a halt.

What we need to do instead is log an error and skip the offending Ingress.